code/modules/html_interface/voting/voting.dm

```diff
@@ -177,6 +185,40 @@ var/global/datum/controller/vote/vote = new()
 		text += "<b>Vote Result: Inconclusive - No Votes!</b>"
 	return text
 
+/datum/controller/vote/proc/weighted_not_previous(var/previous_pick)
+	var/previous_penalty=0.75 // multiplied by the number of votes to reduce (or increase) the chance an item will be picked a second time in a row.
+	var/vote_threshold = 0.15
+	var/list/discarded_choices = list()
+	var/discarded_votes = 0
+	var/total_votes = get_total()
+	var/text
+	var/list/filteredchoices = tally.Copy()
+	var/qualified_votes
+	if (!previous_pick) //if we aren't given the last pick, fall back to weighted voting
+		return weighted()
+
+	if (total_votes > 0)
+		for(var/a in filteredchoices)
+			if(!filteredchoices[a])
+				filteredchoices -= a //Remove choices with 0 votes, as pickweight gives them 1 vote
+				continue
+			if(filteredchoices[a] / total_votes < vote_threshold)
+				discarded_votes += filteredchoices[a]
+				filteredchoices -= a
+				discarded_choices += a
+			if (a==previous_pick) //the penalty is applied after the threshold to ensure it isn't dropped arbitrarily
+				filteredchoices[a]*=previous_penalty
+		if(filteredchoices.len)
+			winner = pickweight(filteredchoices.Copy())
+		qualified_votes = total_votes - discarded_votes
+		text += "<b>Random Weighted Vote Result: [winner] won with [tally[winner]] vote\s and a [round(100*tally[winner]/qualified_votes)]% chance of winning.</b>"
```

Comment:
Might risk a div/0 here - ensure qualified_votes can't ever be 0 or null.

Reply:
0 votes is already accounted for earlier in the code i'm pretty sure.
plus this section is copy+paste from weighted voting

Reply:
yeah, line 200
```
		if (total_votes > 0)
			[...]
		else
		text += "<b>Vote Result: Inconclusive - No Votes!</b>"
	return text